### ISSUE TITLE
fromRequest() always produces quoted strings

### ISSUE THREAD

**Issue by maverick1601** [spring-cloud/spring-cloud-contract#559]
We want to use `fromRequest()` in Contract response to reflect query parameters from the request, e.g.:
`number : fromRequest().query('page')`

However, this always produces quoted string values in the generated stubs instead of reflecting the type from the request, e.g. integral number. It seems like the corresponding response transformer tries to handle everything as string, which of course is more fail-safe.

**Comment by marcingrzejszczak** [comment]
Can you provide more information on the used spring cloud contract version? The DSL, how the generated stub looks like etc?

**Comment by maverick1601** [comment]
Our SCC version is 1.2.1.

In the contract, we have query parameters in the request section, e.g.:
`parameter 'size' : 4`
Ideally, we would want to use a regex here, allowing the user to call us with any integral number. As such, in the response we do not want to use a hard-coded number either, but rather refer to the request, e.g.:
`body([ size : fromRequest().query('size'), ... ])`

This, however, yields to quoted strings in the generated JSON stub, e.g.:
`"size" : "{{{request.query.size.[0]}}}"`

Ideally, fromRequest() would allow us to specify the type of the result, as it cannot guess in advance. As I said, string is fail-safe, but in this case yields wrong results.

**Comment by marcingrzejszczak** [comment]
Ok, now I get it. I think that is possible to be fixed.

**Comment by maverick1601** [comment]
The problem is that `fromRequest()` cannot infer the proper type. IMO there are two solutions: first one could avoid wrapping it in quotes in the stub already, but rather leave it to the response transformer to determine the type. Second one could explicitly set the type in the contract, e.g. `fromRequest().query(...).asType(Double.class)`

**Comment by marcingrzejszczak** [comment]
Yeah, I would rather try going with the 1st option. OTOH maybe the explicit type will need to be introduced for other pieces of the DSL, we'll see. Thanks for the pointer though.

**Comment by marcingrzejszczak** [comment]
You can check the latest snapshots - the problem should be fixed now.

**Comment by acdcjunior** [comment]
Hey, there! (please let me know if it is better to open a new issue instead of commenting on this one).

I see the commit 2fc4137 landed in 1.2.4.RELEASE, and considering [the contract in the test in the commit][test-contract]:

```groovy
Contract.make {
      request {
	      method 'GET'
	      url( '/mytest') {
		      queryParameters {
			      parameter("foo", "bar")
			      parameter("number", 1)
		      }
	      }
	      body("""{ "name": "My name" }""")
      }
      response {
	      status 200
	      body (
			      foo: fromRequest().query("foo"),
			      number: fromRequest().query("number")
	      )
      }
}
```
When I run today, I see the generated asserts are:

```java
assertThatJson(parsedJson).field("['foo']").isEqualTo("bar");
assertThatJson(parsedJson).field("['number']").isEqualTo(1);
```

But when I remove the `body("""{ "name": "My name" }""")` from the contract, the asserts go back to:

```java
assertThatJson(parsedJson).field("['foo']").isEqualTo("bar");
assertThatJson(parsedJson).field("['number']").isEqualTo("1");
```

Also, in either case, the generated mapping (`.json`) contains:

```json
    "body" : "{\"number\":\"{{{request.query.number.[0]}}}\",\"foo\":\"{{{request.query.foo.[0]}}}\"}",
```

Wouldn't it be the case it should skip the quotes in the `number`? That is, generate this instead:

```json
    "body" : "{\"number\":{{{request.query.number.[0]}}},\"foo\":\"{{{request.query.foo.[0]}}}\"}",
```

Thanks!

  [test-contract]: https://github.com/spring-cloud/spring-cloud-contract/commit/2fc413755b46284315bd544619d0c429e7773c29#diff-e964e8e98c8a7a5f1ce070fcef6fe72f2b5ba3af117ea6dee9bd9409ebde7731R386-R404

**Comment by marcingrzejszczak** [comment]
Yeah please file a new issue
